### PR TITLE
Upgrade to Micronaut 3.8.0 and related dependencies

### PR DESCRIPTION
**Pull request by jdpgrailsdev**
## What
* Re-apply https://github.com/airbytehq/airbyte/pull/20716

## How
* Upgrade dependencies to latest Micronaut release and related dependencies

## Recommended reading order
1. `deps.toml`

### CODE REVIEW

File: deps.toml
Comment (benmoriceau) [inline deps.toml:98]: I guess that we will see with the acceptance test but we might change how we call this lib. THis was likely the cause of the issue previously.

Comment (gosusnp) [reply deps.toml:98]: The unknown host exception is the same error I was seeing when I was testing with `3.8.0`.
To clarify what I was saying earlier, I think the issues I saw when I tested`3.7.5` may have been fixed by the changes I merged yesterday.

**Comment by jdpgrailsdev** [comment]
@davinchia After doing some investigation, I discovered that the issue is due to a change in how the AWS SDK generates the URL when requesting S3 objects.  From what I can tell, the client use to use the format `http://<hostname>/<bucket name>`.  Now, it uses the following format:  `<bucket name>.<hostname>`.  This collides with the default DNS generation for K8 services, specifically for our Minio S3 stand-in.  I have tweaked the configuration files to ensure that we can handle the new format and verified that all of the K8 acceptance tests now pass locally.  I also manually created a connection and ran a sync a few times in my local minikube to verify it works.  I could use another set of eyes on this approach to see if there is a better way to do it.

**Review by davinchia — APPROVED** [review]
Changeset looks good.

Should wait for the helm build to pass before merging in.

File: kube/resources/airbyte-minio.yaml
Comment (gosusnp) [inline kube/resources/airbyte-minio.yaml:45]: I think the other error I saw is tied to the config: `STATE_STORAGE_MINIO_BUCKET_NAME`. In our configs, we have two buckets, one for logs, one for states which should to be state of the orchestrator.
I did a quick search, it seems that both are set with the same value `airbyte-dev-logs`. As long as we do not have different values for minio, this should work.

**Review by gosusnp — APPROVED** [review]

File: kube/resources/airbyte-minio.yaml
Comment (jdpgrailsdev) [reply kube/resources/airbyte-minio.yaml:45]: Agreed that this is a risk/something that probably needs to be refactored if/when they diverge.

**Review by benmoriceau — APPROVED** [review]
LGTM, I wonder if we will need something similar in cloud to support their acceptance tests

**Comment by jdpgrailsdev** [comment]
> LGTM, I wonder if we will need something similar in cloud to support their acceptance tests

@benmoriceau @gosusnp brought this up as well.  I am going to test the cloud acceptance tests locally to see if those tests are also impacted.

**Comment by jdpgrailsdev** [comment]
@davinchia I have updated the PR to ensure that we use the older versions of the AWS SDK dependencies to avoid the behavior change in the URL used to access objects.  This should allow us to move forward while we try to figure out how best to re-configure the local Minio pod.

File: airbyte-container-orchestrator/build.gradle
Comment (davinchia) [inline airbyte-container-orchestrator/build.gradle:14]: can we add comments in all the spots where we need to force s3 due to the break for context?

File: airbyte-server/build.gradle
Comment (davinchia) [inline airbyte-server/build.gradle:23]: same comment here

File: airbyte-workers/build.gradle
Comment (davinchia) [inline airbyte-workers/build.gradle:28]: and here

**Review by davinchia — APPROVED** [review]
Good to merge after adding comments!

Should we also create an issue to track this?

**Comment by jdpgrailsdev** [comment]
> Good to merge after adding comments!
> 
> Should we also create an issue to track this?

Definitely.  I have saved the Kubernetes changes off to the side.  I'll create a ticket to track this.

File: airbyte-container-orchestrator/build.gradle
Comment (jdpgrailsdev) [reply airbyte-container-orchestrator/build.gradle:14]: @davinchia An additional comment beyond the one listed that talks about avoiding updates from transitive dependencies?

Comment (davinchia) [reply airbyte-container-orchestrator/build.gradle:14]: yes, that in this case the transitive dependencies from micronaut 3.8 and higher are incompatible with our current helm setup. We could also link to the created issue for better visibility.

**Pull request merged**
